service/matching/matching_engine_test.go

```diff
@@ -475,13 +472,17 @@ func (s *matchingEngineSuite) TestPollWorkflowTaskQueue_GetHistoryFailure() {
 			Query:     &querypb.WorkflowQuery{QueryType: "q"},
 		},
 	}
-	queryDoneCh := make(chan struct{})
+
 	go func() {
-		_, err := s.matchingEngine.QueryWorkflow(context.Background(), &query)
+		ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
+		defer cancel()
+		_, err := s.matchingEngine.QueryWorkflow(ctx, &query)
 		s.ErrorIs(err, fakeErr)
-		queryDoneCh <- struct{}{}
 	}()
 
+	// Wait for query task to be generated before polling for it
+	time.Sleep(20 * time.Millisecond)
```

Comment:
Rather than hard-coding a sleep can we poll it repeatedly until, say, a timeout occurs? I don't trust hard-coded timeouts in tests; they're brittle